src/Shared/Components/ComponentDescriptorSerializer.Deserialize.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text.Json;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components
+{
+    // ComponentDescriptorSerializer is an abstraction shared by both MVC and Blazor
```

Comment:
nit: not an abstraction, it just a thing.

Reply:
Removed the comment, this class has just become ServerComponentSerializationSettings